db-engine/src/main/java/com/splicemachine/db/iapi/types/SQLChar.java

```diff
@@ -2616,57 +2620,24 @@ public ConcatableDataValue substring(
                     SQLState.LANG_SUBSTR_START_OR_LEN_OUT_OF_RANGE);
         }
 
-        // Return null if length is non-positive
-        if (lengthInt < 0)
+        if (lengthInt == 0)
```

Comment:
Can lengthInt be negative?

Reply:
We don't allow length to be negative. It will error out a few lines before the above code:
```
        if ((startInt <= 0 || lengthInt < 0 || startInt > maxLen ||
                lengthInt > maxLen - startInt + 1))
        {
            throw StandardException.newException(
                    SQLState.LANG_SUBSTR_START_OR_LEN_OUT_OF_RANGE);
        }
```